### PR TITLE
CSS Path Relative

### PR DESCRIPTION
**Pull request by darrenburns**
`css_path` is taken to be relative from the app location.

### CODE REVIEW

File: src/textual/file_monitor.py [hunk 1]
```diff
@@ -12,7 +11,7 @@
 class FileMonitor:
     """Monitors a file for changes and invokes a callback when it does."""
 
-    def __init__(self, path: str | PurePath, callback: Callable) -> None:
+    def __init__(self, path: PurePath, callback: Callable) -> None:
```
Comment (darrenburns) [inline src/textual/file_monitor.py:14]: Is this class going to remain part of the public API? If so, I can change this back to give users flexibility.

If it's just going to be internal though I think we should convert all str | PurePath at the API boundary so we have less propagation of unions like this.

**Comment by darrenburns** [comment]
@willmcgugan hold fire on reviewing this, it's not working with `textual run`...

**Comment by darrenburns** [comment]
Seems like it's not an issue with this PR, it's happening for me on `css` branch too. `python just_a_box.py` loads the CSS, `textual run just_a_box.py` doesn't.

File: src/textual/file_monitor.py (hunk 1 above)
Comment (willmcgugan) [reply src/textual/file_monitor.py:14]: This probably should be private. I agree about reducing unions.

**Review by willmcgugan — APPROVED** [review]

**Pull request merged**